backend/graph/api/resolver.go

```diff
@@ -406,3 +411,89 @@ func getProfile(ctx context.Context) (*common.Profile, error) {
 	}
 	return p, nil
 }
+
+func getTask(ctx context.Context, resolver *Resolver, taskID string) (*common.Task, error) {
+	uid, err := uuid.Parse(taskID)
+	if err != nil {
+		return nil, err
+	}
+	task, err := resolver.Loaders.StudyTaskLoader.Get(ctx, uid)
+	if err != nil {
+		return nil, err
+	}
+	if task == nil {
+		return nil, ErrItemNotFound
+	}
+	return task, nil
+}
+
+func getEpisode(ctx context.Context, resolver *Resolver, episodeID string) (*common.Episode, error) {
+	id, err := strconv.ParseInt(episodeID, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	episode, err := batchloaders.GetByID(ctx, resolver.Loaders.EpisodeLoader, int(id))
+	if err != nil {
+		return nil, err
+	}
+	if episode == nil {
+		return nil, ErrItemNotFound
+	}
+	return episode, nil
+}
+
+func (r *Resolver) sendMessage(ctx context.Context, itemID uuid.UUID, message *string, metadata map[string]any) (string, error) {
+	err := ratelimit.Endpoint(ctx, "messages:send:"+itemID.String(), 2, false)
+	if err != nil {
+		return "", err
+	}
+	id, err := utils.GenerateRandomSecureString(32)
```

Comment:
This is the question for the function impl, but under what circumstances can this realistically fail?

Reply:
I'm pretty sure never. Could probably just ignore the error inside the function, as there's no way it will actually receive invalid input

Reply:
In case you input a negative number it will return an error actually.